### ISSUE TITLE
Autorelease @next tag on master commits

### ISSUE THREAD

**Issue by bilby91** [RobinBuschmann/sequelize-typescript#147]
I create this issue to follow the conversation in #143 

The idea would be the following. Every time a new stable release is created. We need to dump the version to the release with beta. So, lets say that we just published 0.6.0. After that publish we need to create a new commit that changes the version to 0.7.0-beta.

When that commit is merged into master, CI will do the following:

```
- git config --global -l
- git config --global user.email circleci@suttna.com
- git config --global user.name CircleCI
- npm version prerelease # (This will bump version to 0.6.0-beta.0, 1, 2, etc) 
- npm publish --tag next # (Tag the version with next so that users can do `npm install sequelize-typescript@next`
- git push --follow-tags # (Push version bump to master with tags)
```

For doing this, the ssh key that is in travis should have write access to the repo. Maybe we can follow this conversation in another commit.

**Comment by RobinBuschmann** [comment]
Hey @bilby91 , thank you for preparing this. This week I don't have time to look into this and give any feedback.

**Comment by bilby91** [comment]
@RobinBuschmann No problem! Let me know! The setup above is working nice for me at the moment so I think it will work!

**Comment by RobinBuschmann** [comment]
Hmm, until now I've tagged the `*-beta.*` releases with `--tag beta`, but I think you're right, `--tag next` seems to be best practice - especially to make `install ...@next` possible.

One question is left from my side: What needs to be done to give travis permission to publish to npm? Is setting the NPM_TOKEN as environment variable enough to achieve this?

**Comment by bilby91** [comment]
@RobinBuschmann Great!

Yes, NPM_TOKEN should be available as an environment variable. You can take a look here for a simple example https://github.com/suttna/botbuilder-slack/blob/master/circle.yml

**Comment by jbender** [comment]
FWIW here are the docs: https://docs.travis-ci.com/user/deployment/npm/

You'll want to add
```yaml
deploy:
  provider: npm
  email: "YOUR_EMAIL_ADDRESS"
  on:
    branch: master
  tag: next
```

then add your encrypted token:
```shell
travis encrypt YOUR_AUTH_TOKEN --add deploy.api_key
```

**Comment by RobinBuschmann** [comment]
@jbender thanks for your input :) but as stated here https://github.com/RobinBuschmann/sequelize-typescript/pull/143#issuecomment-331995668, I will configure it in a different way

**Comment by RobinBuschmann** [comment]
@bilby91 This `npm version prerelease` only increases `*-beta.x`, right? So the new upcoming versions has to be set manually or do I overseen something?

**Comment by bilby91** [comment]
@RobinBuschmann Yep, that is correct. To move to next version a new commit with the bump change needs to be done. Just that

**Comment by RobinBuschmann** [comment]
@bilby91 Can this be closed on favor of #202?

**Comment by bilby91** [comment]
Yes! Make sense. Closing!
